### DISCUSSION TITLE
`@vercel/speed-insights` not collecting data

### DISCUSSION THREAD

**Discussion by michaelaflores** [vercel/community#5125] [Help]
### Summary

I've added a `@vercel/speed-insights` to a Next project and set up the component in my app using page router like so:

```
import { SpeedInsights } from '@vercel/speed-insights/next'

const MyApp = ({ Component, pageProps }: AppProps) => {

  return (
    <>
     <Component {...pageProps} />
    <SpeedInsights />
  )
}

export defaultMyApp
```

I've deployed the project and received visitors, but the dashboard says there's no datapoints even after a few hours. Locally, when I add the `debug` prop, I don't see any data collection reported to the browser console. 

I see the request for the script in the network tab, and no errors are reported to the browser console. The only thing I don't see is the `/_vercel/speed-insights/script.js` actually in the HTML, which is probably the issue here. I see the speed insights script is injected in the HTML <head> `<script src="https://va.vercel-scripts.com/v1/speed-insights/script.debug.js" defer="" data-sdkn="@vercel/speed-insights/next" data-sdkv="1.0.0" data-route="/"></script>`. 

### Steps to Reproduce

1. Install speed insights package
2. Add to NextJS 14.0.3 app
3. Enable debugging prop
4. Observe no collection logged to console

**Comment by michaelaflores** [comment]
This ended up requiring a fix on Vercel's end.

**Comment by lusa9** [comment]
Hi @michaelaflores, can you elaborate more on how the issue was solved since I'm experiencing the same even though I can see events being logged in the console but can't see them in dashboard. 

Much appreciated!

**Reply by michaelaflores** [reply]
Hey @lusa9 – unfortunately I don't have much more info here. When I observed this issue, I opened a case with Vercel Support. Together we verified my setup was proper, and that data was indeed missing. After a day or so, they followed up that they had fixed an issue and data was then flowing properly. I'd recommend opening a support case as it sounds like this bug wasn't just for me...

**Comment by austintoddj** [comment]
Same as @lusa9 - I added SpeedInsights to my project and what was initially working, isn't anymore. No data in the dashboard, but I do see the script included in the page.

**Comment by lusa9** [comment]
I went the same route @michaelaflores did:
- open a case through Vercel Support
- they reach out saying it's a known issue for a small subset of users and ask you to provide the integration snippet
- they do troubleshooting on their end and apply a fix inside 24 hours

@austintoddj let me know if this resolved the issue for you as well. If not, you know the drill ☝️
